wazuh/wazuh-dashboard-plugins
# Fix error when changing API in Agent preview of Endpoints Summary

## Description
This PR resolves the error when changing APIs in Agent preview of Endpoint Summary. To do this, the `HOC withGuard` is added to endpoint summary agent view. HOC withGuard is elevated from AgentWelcome, Inventory data, Stats, and Configuration to AgentView.
An `EuiCallOut` that is no longer necessary is also removed from `main-agent`, since it will not be rendered when there is no pinned agent. This also relocates the `main-agent title` so that it is rendered whenever there is an agent pinned.
Fixed setting breadcrumb when there is no agent information, so that it doesn't add an invalid bread.
Fixed agent selection button on the "No agent is selected" screen by modifying the pinAgent method.
 
## Issues Resolved
- #6799 

## Evidence

### Syscollector

[Evidence_Syscollector.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/0156bc74-c3fc-4a6f-a727-03e7a16c5e43)

### Stats

[Evidence_Stats.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/2eabed32-7d52-4b9f-8a37-3eea2ff41df8)

### Configuration

[Evidence_Configuration.webm](https://github.com/wazuh/wazuh-dashboard-plugins/assets/43619595/42be870e-4990-422f-b2f4-6e2cf07280b0)

## Test

- Navigate to Endpoints Summary.
- Select an agent.
- Go to Syscollector (Inventory data) tab of the agent preview.
- Change the API.
- Check that the "No agent is selected" screen appears and that both the link to Endpoints Summary and the button to select an agent work. 
- Repeat the three last points with Stats and Configuration tabs.

## Check List
